Siil child coverage screen

skostuch opened the pull request:
Changes to Child Coverage Screen as specified in requirements.

benleibert commented:
Hello,

Overall, I think this pull request looks really good – thank you. There are just a few minor things that I think may be worth addressing:

---

Commit 5560afb added the following key/values to messages_en.properties:

+label.vaccination.polio.newborn = VPO (à la naissance)
+label.vaccination.polio.first.dose = VPO 1ère dose
+label.vaccination.polio.second.dose = VPO 2ème dose
+label.vaccination.polio.third.dose = VPO 3ème dose
+label.vaccination.penta.first.dose = Penta (DTC+HepB+Hib) 1ère dose
+label.vaccination.penta.second.dose = Penta (DTC+HepB+Hib) 2ème dose
+label.vaccination.penta.third.dose = Penta (DTC+HepB+Hib) 3ème dose
+label.vaccination.pcv10.first.dose = PCV 13 1ère dose
+label.vaccination.pcv10.second.dose = PCV 13 2ème dose
+label.vaccination.pcv10.third.dose = PCV 13 3ème dose

I believe the values are French, though, and thus exclusively belong in the messages_fr.properties file. (I believe the original English values shown on page 2 of the "Soldevelo SIIL Requirements Final" spec should be used instead.)

---

Commit b6168c3 adds "VAA (Vaccin Anti-Amaril)" as a key-name to vaccinationNamesMap (in child-coverage-controller.js) as well as to the database's "coverage_target_group_products" table. I wonder, though, if "VAA" can be used as the key's name instead. It seems this would more closely match the other keynames (like "BCG" and "IPV").

Commit d71a338 adds a "getSafely" check within ChildCoverageLineItemDTO. I assume this is necessary due to the fields (related to "12-23 months") which have been removed from the Child Coverage screen. To someone who isn't intimately familiar with the project, though, it not clear why the "getSafely" check is necessary within ChildCoverageLineItemDTO but not the analogous AdultCoverageLineItemDTO class. Would you mind, therefore, commenting this within ChildCoverageLineItemDTO, please?

Thank you,
Ben

skostuch commented:
Hi,
I've made the proper fixes as you requested. Following changes have been done:
- added comment on getSafely method
- updated english translations for vaccinations you mentioned
- changed the key from VAA (Vaccin Anti-Amaril) to VAA (the value visible on view is still as it was previously)

benleibert commented:
That looks good – thank you very much!
